src/raster.js

```diff
@@ -224,43 +249,73 @@ const Raster = ({
     if (ext === '.zarr') {
       if (variable) {
         zarr().loadGroup(source, (error, data, metadata) => {
+          if (!Object.keys(data).includes(variable)) {
+            throw new Error(
+              `variable ${variable} not found in zarr dataset, options are: ${Object.keys(
+                data
+              )}`
+            )
+          }
+          if (!bounds && data[lat] && data[lon]) {
+            boundsRef.current = {
+              lat: [
+                data[lat].data.reduce((a, b) => Math.min(a, b)),
+                data[lat].data.reduce((a, b) => Math.max(a, b)),
+              ],
+              lon: [
+                data[lon].data.reduce((a, b) => Math.min(a, b)),
+                data[lon].data.reduce((a, b) => Math.max(a, b)),
+              ],
+            }
+          }
           zarrGroupCache.current = data
           isLoaded.current = true
           texture.current(zarrGroupCache.current[variable])
-          redraw('on zarr group load')
+          redraw.current('on zarr group load')
         })
       } else {
         zarr().load(source, (error, data) => {
           isLoaded.current = true
           texture.current(data)
-          redraw('on zarr array load')
+          redraw.current('on zarr array load')
         })
       }
     }
-  }, [source])
+  }, [source, lat, lon])
 
   useEffect(() => {
     // handle variable change on cached zarr group data
     if (zarrGroupCache.current) {
       texture.current(zarrGroupCache.current[variable])
-      redraw('on variable change')
+      redraw.current('on variable change')
     }
   }, [variable])
 
   useEffect(() => {
-    lut.current({
-      data: colormap,
-      format: 'rgb',
-      shape: [colormap.length, 1],
-    })
-    redraw('on colormap change')
+    boundsRef.current = bounds
```

Comment:
I think we want to only do this conditionally when the `bounds` prop is present. Otherwise, this is at risk for overwriting the calculated value (relevant when `source` data is loaded synchronously).